src/coreclr/jit/importer.cpp

```diff
@@ -4344,19 +4344,23 @@ GenTree* Compiler::impImportCnsTreeFromBuffer(uint8_t* buffer, var_types valueTy
         }
         case TYP_REF:
         {
-            void* ptr;
+            size_t ptr;
             memcpy(&ptr, buffer, sizeof(ssize_t));
 
             if (ptr == 0)
             {
                 tree = gtNewNull();
             }
-            else
+            else if ((ptr & 1) == 0) // Check the lowest bit for "Is frozen" mark
```

Comment:
This should be a method in JIT/EE interface. (I think it is ok for it to be inline method in corinfo.h - not actual JIT/EE interface call.)